docs/reusing-schemas.md

```diff
@@ -0,0 +1,246 @@
+    response(401, "Unauthorized", Schema.ref(:Error))
+  end
+  
+  
+
+  @doc false
+  def swagger_definitions do
+    %{
+        ListOfBooks: ...schema definition...,
+        Error: 
+          swagger_schema do
+            properties do
+              code(:string, "Error code", required: true)
+              message(:string, "Error message", required: true)
+            end
+          end
+      } 
+  end
+end
+```
+
+Our ProjectsController and BooksControllers have now identical `Error` schema defined in their modules.
+
+
+## Extracting schemas into a module for reuse
+
+We can easily extract common schemas into an ordinary elixir module. This module has to implement `PhoenixSwagger`
+behaviour which understands `phoenix_swagger` macros for defining schemas.
+
+```elixir
+defmodule CommonSchemas do
+  @moduledoc "Common schema declarations for phoenix swagger"
+  
+  use PhoenixSwagger
+  
+  @doc """
+  Returns map of common swagger definitions merged with the map of provided schema definitions.
+
+  The common definitions (data structures) are not specific to any controller or
+  business domain logic.
+  """
+  def create_swagger_definitions(%{} = schemas) do
+    Map.merge(
+      %{
+        Error: 
+          swagger_schema do
+            properties do
+              code(:string, "Error code", required: true)
+              message(:string, "Error message", required: true)
+            end
+          end,
+        Errors:
+          swagger_schema do
+              properties do
+                errors(
+                  Schema.new do
+                    title("Errors")
+                    description("A collection of Errors")
+                    type(:array)
+                    items(Schema.ref(:Error))
+                  end
+                )
+              end
+          end          
+      },
+      schemas
+    )
+  end
+end
+```
+
+As you can see, it is also possible to reference other common schemas defined inside the `create_swagger_definitions/1`.
+
+## Reusing the common schemas
+
+Now, instead of defining the `Error` schema in every controller, we can just use `create_swagger_definitions/1`
+inside our controllers.
+
+```elixir
+defmodule ProjectsController do
+  import CommonSchemas, only: [create_swagger_definitions: 1]
+
+  use HelloWeb, :controller
+  use PhoenixSwagger
+
+  swagger_path :index do
+    get "/projects"
+    produces "application/json"
+    parameter("Authorization", :header, :string, "OAuth2 access token", required: true)
+    parameters do      
+      sort_by :query, :string, "The property to sort by"
+      sort_direction :query, :string, "The sort direction", enum: [:asc, :desc], default: :asc
+      company_id :string, :query, "The company id"
+    end
+    response(200, "OK", Schema.ref(:ListOfProjects))
+    response(401, "Unauthorized", Schema.ref(:Error))
+  end
+  
+  
+
+  @doc false
+  def swagger_definitions do
+    create_swagger_definitions(%{
```

Comment:
I think there is a simpler way to include specific schemas by putting each shared schema in its own module and including it in the map:

```elixir
%{
  ListOfProjects: ... schema definitions ...
  Address: MyAppWeb.Schemas.Adress.schema()
}
```

Reply:
Yes this is something that I did from be beginning. But the problem was that along with `Address` you have to include all other general common schemas, that `Address` schema is using. Otherwise they're not gonna appear in generate `swagger.json`.